### DISCUSSION TITLE
Photon updates not working!?

### DISCUSSION THREAD

**Discussion by VonRehberg** [komoot/photon#602] [General]
Hey guys,
I recently setup a nominatim instance in my kubernetes cluster and wanted to test the update functionality.
I found a typo in OSM and fixed it, and a day later the typo was fixed on my nominatim instance as well.

Yesterday I created a photon instance as well to test the fuzzy search. I connected it to my nominatim postgresql and started an update, but unfortunately the fixed typo was not reflected after the update...

Even your public instance still has the typo, although you claim it's up to date with OSM on a minutely base.
Here's the issue:
https://photon.komoot.io/api/?q=Martin%20Luther%20Lirche
Nominatim won't find it, as I fixed Lirche -> Kirche
https://nominatim.openstreetmap.org/search.php?q=Martin%20Luther%20Lirche&format=jsonv2

Is the update not working or is there something else off?

Best regards,
Joschi

**Comment by lonvia** [comment]
The demo instance at photon.komoot.io uses the Graphhopper dumps which are updated weekly. You'll see your changes by next Tuesday or so.

Regarding your local instance, you need to explain a bit more how you run updates for both the Nominatim and photon database.

**Reply by VonRehberg** [reply]
Thanks for your fast reply and clearification :-)

On my local instance I use nominatim 3.7 and the update process is setup on a daily basis from https://planet.openstreetmap.org/replication/day

I then started photon with the following command:
`
CMD java -jar photon-0.3.5.jar -data-dir /photon -host photon-postgresql-service -port 5432 -database nominatim -user nominatim -password mySpecialPassword
`
When Photon was up and running I opened the /nominatim-update path. In the photon logs I could see that it was updating all the ranks, but when it was done, the typo was still persistent.
Anything I need to change on nominatim side, like minutely updates?

**Reply by lonvia** [reply]
You need to change slightly how the Nominatim updates are run. In particular the indexing step needs to be left to Photon. Please have a look at the [example update script](https://github.com/komoot/photon/blob/master/continuously_update_from_nominatim.sh) that we provide. I've just adapted it to work with the new nominatim CLI tool. It's unfortunately pretty much the only documentation we have about continuous updates at the moment.

**Reply by VonRehberg** [reply]
Ah I see. Thanks a lot I will try that.
But does that mean I will have to reimport the planet pbf?

**Reply by lonvia** [reply]
You have to do the Photon import again, I'm afraid, if you want to catch the updates you did. (Although just not bothering about them is probably just fine. It's unlikely that somebody searches for exactly that mistake that was fixed in the small interval you just updated.) Otherwise just switch to the photon update script and it should work fine for future updates. No need to reimport on the Nominatim side.

**Reply by VonRehberg** [reply]
Got it, guess you're right about the missing fixes ;-) I got a bit to concerned about making it perfect.

Now a not too important question:
From what I see so far, for each photon instance, we would need a nominatim instance. 1:1 because of the update process.
If I'm concerned about georeplication and load balancing of instances there is no way to have one nominatim instance, which delivers updates to 2 separate photon instances, right?

**Reply by lonvia** [reply]
No, that would require some evil hacking.

If you want load balancing, it might be better anyway to let ElasticSearch do the heavy lifting. You can run Photon against an external ES instance instead of using the embedded ES (see parameter `-transport-addresses`). You can then set up as many ES nodes as you like and have multiple Photons talk to them. Just be aware that a) you need to run an ES 5.6 and b) you need to be careful to really have only one Photon running the updates.

Oh, and the usual disclaimer: that's how it should work in theory, I've never tried that in practice. ;)

**Reply by VonRehberg** [reply]
Sounds reasonable. Thanks a lot for your help :-)
